packages/split-view/src/SplitView.ts

```diff
@@ -0,0 +1,416 @@
+    public minPos = 0;
+
+    @property({ type: Number, attribute: false })
+    public maxPos = Infinity;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-start' })
+    public isCollapsedStart = this.dividerPosition === 0;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-end' })
+    public isCollapsedEnd = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dragging = false;
+
+    @property({ type: Boolean, reflect: true })
+    public hovered = false;
+
+    @property()
+    public label?: string;
+
+    private offset = 0;
+    private size = 0;
+    private isOver = false;
+    private lastPosition = 0;
+
+    public constructor() {
+        super();
+        this.onPointerDragged = this.onPointerDragged.bind(this);
+        this.onPointerUp = this.onPointerUp.bind(this);
+        this.resize = this.resize.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+        this.getSplitterSize = this.getSplitterSize.bind(this);
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        window.addEventListener('resize', this.resize);
+    }
+
+    public disconnectedCallback(): void {
+        window.removeEventListener('resize', this.resize);
+        super.disconnectedCallback();
+    }
+
+    protected render(): TemplateResult {
+        const dimension = this.vertical
+            ? ORIENTATIONS.vertical
+            : ORIENTATIONS.horizontal;
+
+        return html`
+            <div
+                class="pane"
+                style=${`${dimension}: ${this.dividerPosition}px`}
+            >
+                <slot name="primary"></slot>
+            </div>
+            <div
+                id="splitter"
+                role="separator"
+                aria-label=${ifDefined(this.label || undefined)}
+                tabindex=${ifDefined(this.resizable ? '0' : undefined)}
+                @keydown=${this.onKeyDown}
+            >
+                ${this.resizable
+                    ? html`
+                          <div id="gripper"></div>
+                      `
+                    : html``}
+            </div>
+            <div class="pane secondaryPane">
+                <slot name="secondary"></slot>
+            </div>
+        `;
+    }
+
+    private onPointerMove(event: PointerEvent): void {
+        this.isOver = true;
+        if (this.dragging) {
+            return;
+        }
+        this.updateCursor(event);
+    }
+
+    private onPointerDown(event: PointerEvent): void {
+        if (event.button && event.button !== 0) {
+            return;
+        }
+        if (this.hovered) {
+            if (this.primarySize !== undefined) {
+                return;
+            }
+            window.addEventListener(
+                'pointermove',
+                this.onPointerDragged,
+                false
+            );
+            window.addEventListener('pointerup', this.onPointerUp, false);
```

Comment:
Can you talk to me about why this is on the `window`? Seems like with https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture we'd be able to listen to this event on the same element as the `pointerdown` event.

Reply:
Use setPointerCapture now (fixed in 7be37f0a7a62320e36fcef7575868809b961766e)